functional_tests/scylla_operator/test_functional.py

```diff
@@ -272,3 +274,57 @@ def test_orphaned_services_multi_rack(db_cluster):
 
     db_cluster.wait_for_pods_readiness(pods_to_wait=2, total_pods=len(db_cluster.nodes))
     assert not get_orphaned_services(db_cluster), "Orphaned services were found after decommission"
+
+
+def test_ha_update_spec_while_rollout_restart(db_cluster: ScyllaPodCluster):
+    """
+    Cover the issue https://github.com/scylladb/scylla-operator/issues/410
+    Validate that cluster resources can be updated while the operator StatefulSet is rolling out.
+    - update cluster specification a few time
+    - start rollout restart in parallel with the update
+    - validate that the cluster specification has been updated
+    """
+    to_terminate = threading.Event()
+    value = 1048576
+    errors = []
+
+    def change_cluster_spec():
+        nonlocal value
+        nonlocal errors
+        while not to_terminate.wait(0.1):
+            try:
+                db_cluster.replace_scylla_cluster_value('/spec/sysctls', [f"fs.aio-max-nr={value + 1}"])
+                # increase the value just when the sysctls spec value has been updated - to prevent the situation when
+                # value was increased, but sysctls spec value was not updated
+                value += 1
+            except Exception as error:  # pylint: disable=broad-except
+                errors.append(str(error))
```

Comment:
Maybe we should also log appearing errors?

Reply:
Added debug print in the log despite I think we don't need it here

Reply:
It must not fail.
Also, the place where "it is printed" may not be reached due to some unexpectedly raised exception.

Reply:
In which situation it may be not reached?

Reply:
Gathered errors can be printed in the assertion where it is checked. If test fails on previous lines of code we will not see it.